cpp/src/reshape/byte_cast.cu

```diff
@@ -80,13 +81,20 @@ struct byte_list_conversion {
 
     rmm::device_buffer null_mask = detail::copy_bitmask(input_column, stream, mr);
 
-    return make_lists_column(input_column.size(),
-                             std::move(offsets_column),
-                             std::move(byte_column),
-                             input_column.null_count(),
-                             std::move(null_mask),
-                             stream,
-                             mr);
+    auto result = make_lists_column(input_column.size(),
```

Comment:
FYI: I did a fix for `make_lists_column` in https://github.com/rapidsai/cudf/pull/12370 thus I think this PR is no longer needed.

Reply:
Sorry for another opposite idea: From discussions, we may want to change the strategy to deal with list sanitization. By the new approach, I'll reverse https://github.com/rapidsai/cudf/pull/12370 and call `purge_nonempty_nulls` in the APIs generating output lists column. As such, this PR is still relevant.

Reply:
From our offline discussion, the course of approaches was reversed again: keep calling `purge_nonempty_nulls` in `make_lists_column` (but that factory func will add a boolean flag allowing to bypass calling `purge_nonempty_nulls`).

So please pause this, waiting for more clarification/implementation.

Reply:
FYI: From our final discussion, we will continue to address non-empty nulls in each individual APIs. Please resume this :fire: